### DISCUSSION TITLE
Finding hyperparameters of given model

### DISCUSSION THREAD

**Discussion by Jiya126** [explosion/spaCy#12707] [Help: Coding & Implementations]
How to find the hyperparameters in "en_coreference_web_trf" model
On using
```
nlp = spacy.load("en_coreference_web_trf")
 
doc = nlp(text)

# Print out component names
msg.info("Pipeline components")
for i, pipe in enumerate(nlp.pipe_names):
    print(f"{i}: {pipe}")

    print("###################################################")
    
    gpipe = nlp.get_pipe(pipe)
    hyperparameter = gpipe.get_config()
    print(hyperparameter)
```
It is throwing error:

ℹ Pipeline components
0: sentencizer
###################################################
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-18-61b06e9f4fba>](https://localhost:8080/#) in <cell line: 17>()
     21 
     22     gpipe = nlp.get_pipe(pipe)
---> 23     hyperparameter = gpipe.get_config()
     24     print(hyperparameter)
     25 

AttributeError: 'Sentencizer' object has no attribute 'get_config'

And similarly for other components and models as well

Is there any way to access the hyperparameters of the model we are using?

Also, another thing, what is the difference between "en_coreference_web_trf" and "en_core_web_trf".
What I know is "en_coreference_web_trf" is in experimental branch of spacy and not mentioned in its documentation

**Comment by rmitsch** [comment]
Hey @Jiya126, you can access information about the components' config via `nlp.config["components"]`. 
`en_coreference_web_trf` includes a pretrained coref component - see [here](https://explosion.ai/blog/coref) for more information.

**Reply by Jiya126** [reply]
Thank you, this works
But how can I change these hyperparameters, since it throws me an error stating config file cannot be overwritten

**Reply by Jiya126** [reply]
Also, where could I find documentation for `en_coreference_web_trf`
I'm unable to find that on https://spacy.io/

**Reply by svlandeg** [reply]
Hi, you can find more documentation in the following places:

- https://spacy.io/api/coref
- https://spacy.io/api/span-resolver
- https://spacy.io/api/architectures#Coref
- https://github.com/explosion/projects/tree/v3/experimental/coref
